addons/website_sale_loyalty/controllers/main.py

```diff
@@ -90,8 +90,11 @@ def claim_reward(self, reward, **post):
             if reward_id in rewards:
                 coupon_id = coupon
         redirect = post.get('r', '/shop/cart')
-        if not coupon_id or not reward_id.exists() or reward_id.multi_product:
+        if not coupon_id or not reward_id.exists():
             return request.redirect(redirect)
+        if reward_id.multi_product and 'product_id' in post:
+            request.update_context(product_id=int(post['product_id']))
```

Comment:
```suggestion
        if reward_id.multi_product:
            if 'product_id' in post:
                request.update_context(product_id=int(post['product_id']))
            else:
                request.redirect(redirect)
```
As far as I can see, you enter `_apply_reward` even if no product is chosen :thinking:, and the first reward product will be chosen automatically.

I think that if no product is given, we should probably keep the current situation where no reward is applied.
@dalaOdoo if no specific product is chosen as reward, should we select the first one, or should we keep the previous behavior where nothing happened ?

Reply:
Yes but the claim_reward route is only called when user clicks the claim button in cart. If users doesn't click the claim button and goes to checkout then it will follow the original behavior (of not adding the product to the cart)

Reply:
@Feyensv yest it's sounds logic to me to not apply if no product selected